core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -257,8 +257,8 @@ private[spark] class Executor(
           val serviceTime = System.currentTimeMillis() - taskStart
           val metrics = attemptedTask.flatMap(t => t.metrics)
           for (m <- metrics) {
-            m.executorRunTime = serviceTime
-            m.jvmGCTime = gcTime - startGCTime
+            m.incExecutorRunTime(serviceTime)
```

Comment:
The purpose of this original patch was to just lock down the visibility, so I think it's fine to have -package-private setters in cases where we don't simply increment/decrement. We could also just only have setters (and not inc/dec). Basically as long as they are private, I don't mind. Maybe there is some value to restricting certain ones to inc/dec to provide more safety.